added ability to exclude certain files/folders from the diff process

appblue opened the pull request.

appblue committed:
added option to specify files/dirs exclusion list, and dryrun to list the script

appblue committed:
remove unnecessary NL character when printing script

appblue committed:
changes to exclusion list management, with shortcut for .git folder

appblue committed:
fix for wrong exclusions check

appblue committed:
added additional filter for matching files based on substing in a path

balki commented:
Thank you for working on this.

1. Please share an example/details for how to use the new flags `--exclude` and `--match` ? Are they mutually exclusive ? do they accept glob pattern/regex ?
2. Pleas move other changes (like dry run) to separate PR.

appblue committed:
--dry removed, --match changed to accept regex, README.md update

appblue commented:
I have removed `--dry` option for now (will add it later once this PR will be approved), and modified the `--match` option to accept the list of regex-es instead of plain strings.

The `README.md` file has been updated to reflect new options, and provide example use cases

appblue commented:
Will need to work a bit more on it, as apparently there is a problem with older python versions. Closing the PR for now - we'll come back later

appblue commented:
Will need to work a bit more on it, as apparently there is a problem with older python versions. Closing the PR for now - we'll come back later
